Dockerfile

```diff
@@ -1,5 +1,5 @@
 # Build the clusterlint binary
-FROM golang:1.16 as builder
+FROM golang:1.17 as builder
```

Comment:
also, does this repo need a `.dockerignore` file?

Reply:
> can we update the pipeline to build the binary in pipeline and only copy it in at the end, rather than build it in docker?

Why? Builder pattern is something we consistently use in our codebase. We want the binary to be consistently built each time even if the underlying worker changes. 

> does this repo need a .dockerignore file?
What do you think should be ignored by docker during docker build?  The final image only contains the clusterlint binary, nothing else.

Reply:
nevermind on multi-stage build, you're right on that one! And accordingly, the .dockerignore would only be required if we went the "build-in-CI" route.